packages/schematics/src/add/index_spec.ts

```diff
@@ -1,14 +1,116 @@
-import { Tree } from '@angular-devkit/schematics';
-import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
+import { Tree, HostTree, SchematicsException } from '@angular-devkit/schematics';
+import { SchematicTestRunner, UnitTestTree } from '@angular-devkit/schematics/testing';
 import * as path from 'path';
+import { getFileContent } from '../utility/get-file-content';
+import { setupProject } from '../utility/setup-project';
 
+const PROJECT_NAME = 'foo';
 const collectionPath = path.join(__dirname, '../collection.json');
 
-xdescribe('add', () => {
-  it('works', () => {
+describe('ng add @clr/angular', () => {
+  it('should print an error if executed in an empty directory', async () => {
     const runner = new SchematicTestRunner('schematics', collectionPath);
-    const tree = runner.runSchematic('add', { module: 'app' }, Tree.empty());
+    try {
+      await runner.runSchematicAsync('ng-add', { module: 'app' }, Tree.empty()).toPromise();
+      fail('expected to throw');
+    } catch (e) {
+      expect(e instanceof SchematicsException);
+      expect(e.message).toEqual('Could not install Clarity, requires Angular and Angular CLI version 6 or greater');
+    }
+  });
+
+  describe('in a new Angular workspace', async () => {
+    let runner: SchematicTestRunner;
+    let workspaceTree: UnitTestTree;
+
+    beforeEach(async () => {
+      runner = new SchematicTestRunner('schematics', collectionPath);
+      workspaceTree = await setupProject(new UnitTestTree(new HostTree()), runner, PROJECT_NAME);
+    });
+
+    it('should add dependencies to Clarity packages', async () => {
+      const tree = await runner.runSchematicAsync('ng-add', {}, workspaceTree).toPromise();
+
+      const packageJsonPath = '/package.json';
+      expect(tree.files).toContain(packageJsonPath);
+      const packageJson = JSON.parse(getFileContent(tree, packageJsonPath));
+      const { dependencies } = packageJson;
```

Comment:
Could this assign the JSON.parse output directly here and elminiate packageJson?